src/foam/lib/parse/BlobPStream.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.lib.parse',
+  name: 'BlobPStream',
+  implements: [
+    'foam.lib.parse.PStream'
+  ],
+  properties: [
+    {
+      class: 'Blob',
+      name: 'blob'
+    },
+    {
+      class: 'Int',
+      name: 'pos_'
+    },
+    {
+      class: 'String',
+      name: 'head_',
+      javaFactory: `
+        java.io.OutputStream os = new java.io.ByteArrayOutputStream(1);
```

Comment:
There must be a faster way to do this, but I can't figure out what it is pre-breakfast.  Will think about it.

Reply:
Haha yeah we'll figure something out. Opportunity for a performance tweak for sure.